language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -387,29 +426,28 @@ impl CompiledPackage {
             .join(CompiledPackageLayout::BuildInfo.path());
 
         // Compare the digest of the package being compiled against the digest of the package at the time
-        // of the last compilation to determine if we can reuse the already-compiled package or not.
+        // of the last compilation to determine if we can reuse the already-compiled package or
+        // not. If any dependency has changed we need to recompile.
         if let Ok(package) = OnDiskCompiledPackage::from_path(&path) {
-            if !(package.has_source_changed_since_last_compile(&resolved_package)
-                || package.are_build_flags_different(&resolution_graph.build_options)
-                // force root package recompilation in test mode
-                || resolution_graph.build_options.test_mode && is_root_package)
+            if dependencies
```

Comment:
IIUC a module don't need to be recompiled iff all its dependencies are unmodified. Is it because we can't do compilation with some modules in cache and some modules modified? As a result whenever there's a dependency change we need to recompile everything?

Reply:
Correct. In the package system we don't look at what functions a given module may be linking against. The only thing we know is that if a module `M` is in package `A`, then `M` could possibly be linked against any modules in `A` or `A`'s dependencies. So if any package in `A`'s dependencies change we need to invalidate `A` and recompile.